app/controllers/course/forum/topics_controller.rb

```diff
@@ -0,0 +1,97 @@
+class Course::Forum::TopicsController < Course::Forum::Controller
+  include Course::ForumTopicsHideAndLockConcern
+
+  before_action :load_topic, except: [:new, :create]
+  load_resource :topic, class: Course::Forum::Topic.name, through: :forum, only: [:new, :create]
+  authorize_resource :topic, class: Course::Forum::Topic.name, through: :forum
+
+  def show
+    @posts = @topic.posts
+    @topic.views.create(user: current_user)
+  end
+
+  def new
+    # When create a topic, a post with the same title will be created too.
+    @post = @topic.posts.new
+  end
+
+  def create
+    @post = Course::Discussion::Post.new
```

Comment:
I'm sorry if this was from v1, but I think you should think carefully when you port it to v2...